docs/en/introduction/architecture.md

```diff
@@ -9,43 +9,64 @@ The JuiceFS file system consists of three parts:
 
-Hence, you cannot find the original files directly in the object storage, instead there's only a `chunks` folder and a bunch of numbered directories and files in the bucket, don't panic, this is exactly how JuiceFS formats and stores data. At the same time, file and its relationship with chunks, slices, blocks will be stored in the metadata engine. This decoupled design is what makes JuiceFS a high-performance file system.
+![Split slices to blocks](../images/slice-to-block.svg)
 
-![](../images/how-juicefs-stores-files.svg)
+After writing a file to JuiceFS, you cannot find the original file directly in the object storage. Instead, the storage bucket contains a `chunks` folder and a series of numbered directories and files. These numerically named object storage files are the blocks split and stored by JuiceFS. The mapping between these blocks, chunks, slices, and other metadata information (such as file names and sizes) is stored in the metadata engine. This decoupled design makes JuiceFS a high-performance file system.
 
-Some other technical aspects of JuiceFS storage design:
+![How JuiceFS storesfiles](../images/how-juicefs-stores-files.svg)
 
-* Files (any size) are not merged and stored. This is for performance considerations and to avoid read amplification.
-* Provides strong consistency guarantee, but can be tuned for performance in different scenarios, e.g. deliberately adjust metadata cache policies, to trade consistency for performance. Learn more at [Metadata cache](../guide/cache_management.md#metadata-cache).
-* Support ["Trash"](../security/trash.md) functionality, and enabled by default. Deleted files are kept for a specified amount of time, to help you avoid data loss caused by accidental deletion.
+Regarding logical data structures, if a file is not generated through continuous sequential writes but through multiple append writes, each append write triggers a `flush` to initiate the upload, resulting in multiple slices. If the data size for each append write is less than 4 MB, the data blocks eventually stored in the object storage are smaller than 4 MB blocks.
+
+![Small append writes](../images/small-append.svg)
+
+Depending on the writing pattern, the arrangement of slices can be diverse:
+
+- If a file is repeatedly modified in the same part, it results in multiple overlapping slices.
+- If writes occur in non-overlapping parts, there will be gaps between slices.
+
+However complex the arrangement of slices may be, when reading a file, the most recent written slice is read for each file position. The figure below illustrates this concept: while slices may overlap, reading the file always occurs "from top to bottom." This ensures that you see the latest state of the file.
```

Comment:
Not sure whether "This ensures that you see the latest state of the file." is correct.
@timfeirg PTAL